pdf_build_src/process_markdowns.py

```diff
@@ -141,6 +142,194 @@ def modify_changelog():
+        print('    - Number of chars max in column (PDF): {}'.format(nb_of_chars_in_pdf))
+        print('    - Final number of chars in first column: {}'.format(first_column_width))
+        print('    - Final number of chars in second column: {}'.format(second_column_width))
+
+    # Format the lines with correct number of dashes or whitespaces and 
+    # correct alignment of fences and populate the new table (A List of str)
+    new_table = []
+    for i, row in enumerate(table):
+
+        if i == 1:
+            str_format = ' {:-{align}{width}} '
+        else:
+            str_format = ' {:{align}{width}} '
+
+        row_content = []
+        for j, elem in enumerate(row):
+            # Set the column width
+            column_width = max_chars_in_cols[j]
+            if j == 1:
+                column_width = first_column_width
+            elif j == 2:
+                column_width = second_column_width
+            
+            if j == 0 or j == len(row) - 1:
+                row_content.append(elem)
+            else:
+                if '`' in elem:
+                    str_format = ' {:{align}{width}} '
+                    row_content.append(str_format.format(elem, align='<', width=(column_width)))
+                elif '-:' in elem and ':-' in elem :
+                    str_format = ' {:-{align}{width}}: '
+                    row_content.append(str_format.format(':-', align='<', width=(column_width)))
+                elif not '-:' in elem and ':-' in elem :
+                    str_format = ' {:-{align}{width}} '
+                    row_content.append(str_format.format(':-', align='<', width=(column_width)))
+                elif '-:' in elem and not ':-' in elem :
+                    str_format = ' {:-{align}{width}}: '
+                    row_content.append(str_format.format('-', align='<', width=(column_width)))
+                elif i == 1 and not '-:' in elem and not ':-' in elem :
+                    str_format = ' {:-{align}{width}} '
+                    row_content.append(str_format.format('-', align='<', width=(column_width)))
+                else:
+                    row_content.append(str_format.format(elem, align='<', width=(column_width)))
+        if debug:
+            print(row_content)
+
+        new_table.append(row_content)
+        
+    return new_table
+
+
+def correct_tables(root_path):
+    """Change tables in markdown files for correct rendering in PDF.
+
+    This modification makes sure that the proportion and number of dashes (---) are 
+    sufficiently enough for correct PDF rendering and fences (|) are corrected aligned.
+
+    Parameters
+    ----------
+    root_path : str
+        Path to the root directory containing the markdown files
+
+    """
+    markdown_list = []
+    for root, dirs, files in os.walk(root_path):
+        for file in files:
+            if file.endswith(".md") and file != 'index.md' and file != '01-contributors.md' and file != '04-entity-table.md':
```

Comment:
As I edited in https://github.com/bids-standard/bids-specification/pull/560#issuecomment-669235630, I think we can go over the entity table now.

BTW: a cleaner syntax would be:

```Python
exclude_files = [a, b, c]
if file.endswith('.md') and file not in exclude_files:
    pass
```